test/test_ops.py

```diff
@@ -194,7 +194,11 @@ def to_cpu(arg):
             self.assertEqual(len(cuda_results), len(cpu_results))
 
             for cpu_result, cuda_result in zip(cpu_results, cuda_results):
-                self.assertEqual(cpu_result, cuda_result)
+                if not op.deterministic and isinstance(cpu_result, torch.Tensor):
+                    self.assertEqual(cpu_result.dtype, cuda_result.dtype)
+                    self.assertEqual(cpu_result.shape, cuda_result.shape)
```

Comment:
@mruberry @amjames here is what I was thinking for an `op. deterministic` flag. In this context the asserts above like `self.assertEqual(len(cuda_results), len(cpu_results))` make more sense. 

We can do a more fine grained check for both `deterministic=True` and `deterministic=False`

Reply:
Shape/dtype checks seem sufficient.